src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -1038,8 +1105,19 @@ private function merge_style_property( string $style_attribute_value, string $st
 	 */
 	private function data_wp_text_processor( WP_Interactivity_API_Directives_Processor $p, string $mode ) {
 		if ( 'enter' === $mode ) {
-			$attribute_value = $p->get_attribute( 'data-wp-text' );
-			$result          = $this->evaluate( $attribute_value );
+			$entries     = $this->get_directive_entries( $p, 'text' );
+			$valid_entry = null;
+			// Get the first valid `data-wp-text` entry without suffix or unique ID.
+			foreach ( $entries as $entry ) {
+				if ( null === $entry['suffix'] && null === $entry['unique_id'] && ! empty( $entry['value'] ) ) {
+					$valid_entry = $entry;
+					break;
+				}
+			}
+			if ( empty( $valid_entry ) ) {
```

Comment:
This may be wrong. We should check if `$valid_entry['value']` is empty, not the whole entry.

Reply:
I think there could be some cases where there are no valid `data-wp-text` directives, which would result into `$valid_entry` being `null`. Additionally, it is only populating `$valid_entry` when `value` is not empty, so I believe it is working as expected. But I may be wrong.

Reply:
Oh, I misunderstood the code. The check is fine. I would simply change the condition to explicitly check if `$valid_entry` is `null`.


```suggestion
			if ( null === $valid_entry ) {
```